MagicModules/TemplateMagicModulesInput.ts

```diff
@@ -185,68 +185,64 @@ function appendUxOptions(output: string[], options: ModuleOption[], prefix: stri
         }
         else
         {
-            switch (option.Type)
-            {
-                case "str":
-                    if (option.NameSwagger == "resourceGroupName")
-                    {
-                        dataType = "!ruby/object:Api::Azure::Type::ResourceGroupName";
-                    }
-                    else if ((typeof option.ExampleValue == "string") && option.ExampleValue.startsWith('/subscriptions/'))
-                    {
-                        dataType = "!ruby/object:Api::Azure::Type::ResourceReference";
-                    }
-                    else if (option.NameSwagger == "location")
-                    {
-                        dataType = "!ruby/object:Api::Azure::Type::Location";
-                    }
-                    else
-                    {
-                        dataType = "!ruby/object:Api::Type::String";
-                    }
-                    break;
-                case "dict":
-                    if (option.IsList)
-                    {
-                        // for now it's the same
-                        dataType = "!ruby/object:Api::Type::NestedObject";
-                    }
-                    else
-                    {
+            if (option.IsList) {
+                dataType = "!ruby/object:Api::Type::Array";
+            } else {
+                switch (option.Type)
```

Comment:
I think you could use the same function you created here as well?
  getItemTypeForList(option: ModuleOption): string